cmd/src/users_prune.go

```diff
@@ -58,65 +60,119 @@ query getCurrentUser {
 }
 `
 		var currentUserResult struct {
-			Data struct {
-				CurrentUser struct {
-					Username string
+			CurrentUser struct {
+				Username string
+			}
+		}
+		if ok, err := cfg.apiClient(apiFlags, flagSet.Output()).NewRequest(currentUserQuery, nil).Do(context.Background(), &currentUserResult); err != nil || !ok {
+			return err
+		}
+
+		// get total users to paginate over
+		totalUsersQuery := `
+query getTotalUsers {
+	site {
+		users {
+			totalCount
+		}
+	}
+}`
+		var totalUsers struct {
+			Site struct {
+				Users struct {
+					TotalCount float64
 				}
 			}
 		}
-		if ok, err := cfg.apiClient(apiFlags, flagSet.Output()).NewRequest(currentUserQuery, nil).DoRaw(context.Background(), &currentUserResult); err != nil || !ok {
+		if ok, err := cfg.apiClient(apiFlags, flagSet.Output()).NewRequest(totalUsersQuery, nil).Do(context.Background(), &totalUsers); err != nil || !ok {
 			return err
 		}
 
+		// get 100 site users
 		getInactiveUsersQuery := `
-query getInactiveUsers {
+		query getInactiveUsers($limit: Int $offset: Int) {
 	site {
 		users {
-			nodes {
+			nodes (limit: $limit offset: $offset) {
+				id
 				username
 				email
 				siteAdmin
 				lastActiveAt
+				deletedAt
 			}
 		}
 	}
 }
 `
 
-		var usersResult struct {
+		// paginate through users
+		var aggregatedUsers struct {
```

Comment:
Can't this just be a slice now? `[]SiteUser`?

Reply:
yes ✅ -- I was stuck in graphql mode